demosplan/DemosPlanCoreBundle/Logic/Statement/StatementCopier.php

```diff
@@ -508,25 +507,27 @@ public function copyStatementObjectWithinProcedureWithRelatedFiles(
         bool $createReport = true,
         bool $copyOnCreateStatement = false,
     ): Statement {
+        if ($statement->getFiles()) {
+            $newStatement = $this->copyStatementObjectWithinProcedure(
+                $statement,
+                $createReport,
+                $copyOnCreateStatement,
+                false // do not flush to avoid constraints at this point
+            );
 
-        $newStatement = $this->copyStatementObjectWithinProcedure(
-            $statement,
-            $createReport,
-            $copyOnCreateStatement,
-            false //do not flush to avoid constraints at this point
-        );
-
-        if (!$statement instanceof Statement) {
-            throw new CopyException('error on copying original statement');
-        }
-
-        // persist to get an ID for the FileContainer copying below
-        $this->getDoctrine()->getManager()->persist($newStatement);
+            if (!$statement instanceof Statement) {
+                throw new CopyException('error on copying original statement');
+            }
 
+            // persist to get an ID for the FileContainer copying below
+            $this->getDoctrine()->getManager()->persist($newStatement);
 
-        $this->statementService->addFilesToCopiedStatement($newStatement, $statement->getId());
+            $this->statementService->addFilesToCopiedStatement($newStatement, $statement->getId());
 
-        return $newStatement;
+            return $newStatement;
+        } else {
+            return $this->copyStatementObjectWithinProcedure($statement);
```

Comment:
Why didn't you pass `$createReport` and `$copyOnCreateStatement`?

Reply:
I thought, they have already defaulted values. 
done in ([404d7ec](https://github.com/demos-europe/demosplan-core/pull/2515/commits/404d7ec2692b2af591f0bbf981fa7be9c62426d3))